File: src/crewmate.py
import numpy as np

from room import Room
from skeld_structure import SkeldStructure


class Crewmate:
    def __init__(self, actual_room):
        self.actual_room = actual_room
        self.move_timer = np.random.randint(1, 4)

    def take_action(self, map):
        if self.move_timer == 0:
            neighbors = map.get_neighbors(self.actual_room)
            move_to = np.random.choice(neighbors)
            self.actual_room = move_to
            self.move_timer = np.random.randint(1, 4)
        else:
            self.move_timer -= 1

def main():
    pass



if __name__ == "__main__":
    main()

File: SearchAgent/SearchAgent/Crewmate.cpp
#include "Crewmate.h"

Crewmate::Crewmate() {
    this->actualRoom = 0;
    this->moveTimer = 0;
}

Crewmate::Crewmate(int actualRoom) {
    this->actualRoom = actualRoom;
    this->moveTimer = 0;// 1 + rand() % 4;
}

int Crewmate::getRoom() const {
    return this->actualRoom;
}

int Crewmate::getMoveTimer() const {
    return this->moveTimer;
}

bool Crewmate::isEliminated() const {
    return this->actualRoom == -1;
}

void Crewmate::eliminate() {
    this->actualRoom = -1;
}

void Crewmate::takeAction(const SkeldStructure& map) {
    if (this->moveTimer == 0) {
        std::vector<int> neighbors = map.getNeighbors(this->actualRoom);
        this->actualRoom = neighbors[rand() % neighbors.size()];
        this->moveTimer = 1 + rand() % 4;
    }
    else {
        this->moveTimer--;
    }
}
